### ISSUE TITLE
cmi.colors.(publicmessage,nickname etc.).hex or .colors issue.

### ISSUE THREAD

**Issue by AutumnUwU** [Zrips/CMI#5625]
**Description of issue:**
I have tried **several** times and have spent several hours into trying to make it so cmi.colors works but it seems that when i do for example: cmi.colors.nickname.hex it'll do nothing, same if i type just cmi.colors.nickname.color or cmi.colors.nickname.colors, i've tried not even joking over 200 combinations, Also when i try doing cmi.colors.nickname.* for example and deny cmi.colors.nickname.hex it allows hex still, And when i try cmi.colors.nickname.hex alone, It allows hex on everything. Please fix this issue. Its as if it doesn't matter the input in cmi.colors.(here). If you put hex it'll let you do hex on everything its very uncustomizable from that, If you need to see my luckperms editior and everything i've done and tried i can send it to you in a private message. I've had this issue for a couple months though

---

---

**Cmi Version (using`/cmi version`):** 
8.8.4.6
**Server Type (Spigot/Paperspigot/etc):** 
Tunity 
**Server Version (using `/ver`):** 
1.16.5-R0.1-SNAPSHOT
**Relevant plugins (Delete if this isn't needed):**
ChatColor2 https://www.spigotmc.org/resources/chatcolor.22692/
LuckPerms latest version as of at the moment.

**Comment by ZEROOblivious** [comment]
I'm having the same issue on my server. 
![image](https://user-images.githubusercontent.com/32251194/120128622-778b4780-c190-11eb-8d66-16bcea89f8c2.png)

![image](https://user-images.githubusercontent.com/32251194/120128641-7fe38280-c190-11eb-8c5f-25232c8abd83.png)

I believe there's an issue with the plugin listening to HEX-Codes. 
All of its enabled on my server. 
They have cmi.colors.nickname.*

**Comment by ItLucifer-MRPG** [comment]
From what Ive seen, its an issue with the alias, /cmi nick works fine with hex colours. This is currently the same with any command you might wanna use hex colours with, just use /cmi [command] for now

**Comment by ZEROOblivious** [comment]
I'd rather wait for a patch, I don't like my players knowing my plugins so I've set custom-alias's on them all so they won't figure them out so having a community doing /cmi nick is not my cup of tea.

**Comment by ItLucifer-MRPG** [comment]
Yeah, I was thinking about doing the same thing, But sadly until its fixed, thats how it is. Atleast there is a way and it isnt fully broken

**Comment by CloudeLecaw** [comment]
have you tried last version 8.8.4.7?

**Comment by AutumnUwU** [comment]
To all comments above: Yes, I've updated to the latest version already, It seems that the issue with /nick is fixed, Thank you :), I still am unsure if changing the configuration will work though, Nickname was already set to true, Haven't tried stuff on signs etc. though yet

**Comment by AutumnUwU** [comment]
It does not seem that signs or books work when I give a player permission to &a-&e or &0-&9, Same with .color or .colors, I am giving them perms such as 
cmi.colors.books.a
cmi.colors.books.color and .colors neither worked.
cmi.colors.signs.a 
cmi.colors.signs.color and .colors neither worked here as well.

Possible to try and fix that?

**Comment by AutumnUwU** [comment]
Certain players also have cmi.colors.signs.hex etc. but trying to write hex in chat doesn't work. Is it possible I am writing the wrong permission? My config replicates what CloudeLecaw sent to me.

**Comment by CloudeLecaw** [comment]
Your a bit wrong....
There is cmi.book.colors
Not cmi.colors.books

**Comment by AutumnUwU** [comment]
I have tried cmi.colors.signs.c, cmi.signs.colors or whatever u said etc. lol, I'll try the newest edit you made.

**Comment by CloudeLecaw** [comment]
its written in the config that you should use the names^^

**Comment by Momshroom** [comment]
Sorry, I mistakenly thought this ticket had been closed and made my very similar report on a new ticket #5653

**Comment by CloudeLecaw** [comment]
Normaly you have these permission you wrote above ending with .hex

**Comment by AutumnUwU** [comment]
> Normaly you have these permissions you wrote above ending with .hex (for all the permissions) thus this case the word hex is the colorname.
> If ubset them ppl are ablento use all hex colors.if you let cmi handle it and set the config we discussed earlier allready. And yes your right thats the differnce with hex. Here u also can give just permission to single hexcolors too if wanted.

I did exactly those permissions like I said but people still do not have perms to hex without *. And i don't want to set single colors, However single colors do work but it'd take a while to add every single hex one by one.

**Comment by Zrips** [comment]
It looks more an issue with permission usage, it should contain actual color name or hex code representing it, so valid ones would be
cmi.colors.publicmessage.red
cmi.colors.publicmessage.cancan
cmi.colors.publicmessage.6600cc
While giving access to all hex color codes isint best solution as it will only invite trouble from people using some extreme looking colors. Its not perfect to give one permission for each color, but its some what better than giving access to all of them. Its using old format which was made to handle original 16 colors, but after we got access to all 16+ million of them in basic hex format and over 1000 named ones, that became some what complicated and messy. 
We did had previously cmi.colors.publicmessage.* type permission nodes, but those caused some issues when trying to negate some colors and giving one with wild card, so it got disabled.
Will make some changes to reenable wild card type permissions for colors so it should resolve your issues.
